_data-prepper/pipelines/configuration/processors/mutate-event.md

```diff
@@ -0,0 +1,306 @@
+
+### Configuration
+
+You can configure the `rename key` processor with the following options.
+
+Option | Required | Description
+:--- | :--- | :---
+| `entries` | Yes | A list of entries to rename in an event. | 
+| `from_key` | Yes | The key of the entry to be renamed. |
+| `to_key` | Yes | The new key of the entry.
+| `overwrite_if_to_key_exists` | No | When set to a value of`true`, if `to_key` already exists in the event, then the existing value will be overwritten. The default value is `false`.
+
+### Basic usage
+
+To get started, create the following `pipeline.yaml` file:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_keys:
+        entries:
+        - from_key: "message"
+          to_key: "newMessage"
+          overwrite_if_to_key_exists: true
+  sink:
+    - stdout:
+```
+
+Create a log file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with the path of this file.
+
+```json
+{"message": "value"}
+```
+
+When run, the processor parses the message into the following output:
+
+```json
+{"newMessage": "value"}
+```
+
+> If `newMessage` already exists, its existing value is overwritten with `value`.
+
+### Special considerations
+
+The renaming operation occurs in a defined order. This means that chaining is implicit with the `RenameKey` processor. See the following `piplines.yaml` file example:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_key:
+        entries:
+        - from_key: "message"
+          to_key: "message2"
+        - from_key: "message2"
+          to_key: "message3"
+  sink:
+    - stdout:
+```
+
+Add the following contents to the `logs_json.log` file:
+
+```json
+{"message": "value"}
+```
+
+After the processor runs, the following output appears:
+
+```json
+{"message3": "value"}
+```
+
+## ConvertEntry
+
+The `ConvertEntry` processor converts a value type associated with the specified key in a message to the specified type. It is a casting processor that changes the types of some fields in the event or message. Some of inputted data may need to be converted to different types, such as an integer or a double, so that it will pass the events through condition-based processors, or to perform conditional routing.
+
+### Configuration
+
+You can configure the `ConvertEntry` processor with the following options.
+
+Option | Required | Description
+:--- | :--- | :---
+| `key`| Yes | Keys whose value needs to be converted to a different type. | 
+| `type` | No | Target type for key value. Possible values are `integer`, `double`, `string`, and `boolean`. Default value is `integer`.|
+
+### Basic usage
+
+To get started with type conversion processor using Data Prepper, create the following `pipeline.yaml` file:
```

Comment:
What does the conversion processor do? Is there a page we can link to for these processors to give users more detail on what it is and why they should use it and how it works?

Reply:
The introduction on Line 257 looks good. I don't have more to add on this.